test/control_filters/test_low_pass_filter.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2023, Stogl Robotics Consulting UG (haftungsbeschränkt)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "test_low_pass_filter.hpp"
+
+TEST_F(LowPassFilterTest, TestLowPassWrenchFilterAllParameters)
+{
+    std::shared_ptr<filters::FilterBase<geometry_msgs::msg::WrenchStamped>> filter_ =
+        std::make_shared<control_filters::LowPassFilter<geometry_msgs::msg::WrenchStamped>>();
+
+    // should allow configuration and find parameters in sensor_filter_chain param namespace
+    ASSERT_TRUE(filter_->configure("", "TestLowPassFilter",
+        node_->get_node_logging_interface(), node_->get_node_parameters_interface()));
+
+    // change a parameter
+    node_->set_parameter(rclcpp::Parameter("sampling_frequency", 500.0));
+    // accept second call to configure with valid parameters to already configured filter
+    ASSERT_TRUE(filter_->configure("", "TestLowPassFilter",
+        node_->get_node_logging_interface(), node_->get_node_parameters_interface()));
+}
+
+
+TEST_F(LowPassFilterTest, TestLowPassWrenchFilterMissingParameter)
+{
+    std::shared_ptr<filters::FilterBase<geometry_msgs::msg::WrenchStamped>> filter_ =
+        std::make_shared<control_filters::LowPassFilter<geometry_msgs::msg::WrenchStamped>>();
+
+    // should deny configuration as sampling frequency is missing
+    ASSERT_FALSE(filter_->configure("", "TestLowPassFilter",
+        node_->get_node_logging_interface(), node_->get_node_parameters_interface()));
+}
```

Comment:
not needed when using generate_parameter_library

Reply:
I don't see why one should not test for `filter configure` failing "properly". This test permits to verify that the parameters were correctly defined in the `generate_parameter_library` YAML as "should fail if missing", and that the throw is caught correctly isn't it ?